enhancements/api-v2.md

```diff
@@ -0,0 +1,140 @@
+---
+title: Quay API V2 design
+authors:
+  - "@sdadi"
+reviewers:
+  - "@syahmed"  
+  - "@sleesinc"
+  - "@obulatov"
+  - "@bcaton"
+approvers:
+  - "@syahmed"  
+  - "@sleesinc"
+  - "@obulatov"
+  - "@bcaton"
+creation-date: 2023-07-24
+last-updated: yyyy-mm-dd
+status: implementable
+---
+
+# Quay API v2
+
+This is a proposal to design a generic strategy to improve Quay API's to support pagination, filtering and sorting.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+Currently, only a few APIs have the option to paginate results on the backend. The goal of the proposal is to have a re-usable system to perform actions on the backend and send the curated result on the frontend. Ideally frontend should only display the results and allow the backend to perform
+all the heavy lifting.
+
+## Motivation
+
+In the quay ui, filtering, sorting and occasionally paginating the results happen on the frontend. 
+This leads to high rendering time on the frontend and thereby a poor user experience.
+
+### Goals
+
+Implement a generic system that is consistent across different endpoints to support the following operations on Quay endpoints:
+- pagination
+- filtering
+- sorting
+
+## Design Details
+
+### API Design
+
+- Proposed path of the endpoints: In an attempt to not change the existing endpoints, new APIs will be under `/api/v2_draft/`.
+  `v2_draft` will be replaced with `v2` when ready to release. 
+- Existing decorators like the scope of API, page parameters will be reused where ever required.
```

Comment:
Is this not going to be backward-compatible? If just adding pagination to the current existing resources.